.github/workflows/python-app.yml

```diff
@@ -12,8 +12,28 @@ on:
     types: [published]
 
 jobs:
+  version:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
+    steps:
+      - name: Checkout main branch
+        uses: actions/checkout@v3
+        with:
+          ref: 'main'
+          ssh-key: ${{ secrets.SSH_KEY }}
+      - name: Set version
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+      - name: Update version
+        run: python scripts/update_version.py ${{ env.RELEASE_VERSION }}
+      - name: Auto-commit changes
+        uses: stefanzweifel/git-auto-commit-action@49620cd3ed21ee620a48530e81dba0d139c9cb80
```

Comment:
why the specific hash?

Reply:
To mitigate risk. I think it's a safer play since we're using a third-party action here

Reply:
do you know if devs can overwrite a version name? in any case the commit hash it's more certain

Reply:
Agreed on the certainty^ And yes, I believe devs can replace the tag in git